src/app/shared/expiration/date-picker/expiration-date.component.ts

```diff
@@ -19,14 +19,24 @@ export class ExpirationDateComponent implements OnInit, OnDestroy {
   /**
    * Defined in milliseconds.
    */
-  @Input() defaultValidityPeriod: number;
+  @Input() set defaultValidityPeriod(period: number) {
+    if (!period) {
+      return;
+    }
+    this._defaultPeriod = period;
+    this.defaultExpirationDate();
```

Comment:
Maybe this is a side point, but I find understanding the difference between `defaultExpirationDate` and `setDefaultExpirationDate` challenging. It seems like `setDefaultExpirationDate` has the same code as `defaultExpirationDate`, maybe `setDefaultExpirationDate` could call `defaultExpirationDate` and the function names could be adjusted accordingly?

Reply:
ahh, yeah. Code looks similar. I will adjust it, thanks for pointing this out.

Reply:
https://github.com/energywebfoundation/switchboard-dapp/pull/783 hope this is what you wanted :)

Reply:
Thanks!